src/identity.rs

```diff
@@ -222,12 +222,19 @@ impl Password {
         S3: Into<String>,
     {
         let url = auth_url.into_url()?;
-        // TODO: more robust logic?
-        let token_endpoint = if url.path().ends_with("/v3") {
-            format!("{}/auth/tokens", url)
+
+        let token_endpoint = if url.path().ends_with("/") {
```

Comment:
I think it can be done a bit nicer with something like
```rust
let mut url = auth_url.into_url()?;
url.path_segments_mut().map_err(|_| <figure out something>).pop_if_empty();
```